hubblestack/modules/fdg.py

```diff
@@ -256,20 +279,20 @@ def top(fdg_topfile='top.fdg'):
     for fdg_file in fdg_routines:
         if isinstance(fdg_file, dict):
             for key, val in fdg_file.items():
-                retkey, retval = run(key, val)
+                retkey, retval = call_function(key, val, fdg_version=fdg_version)
                 ret[retkey] = retval
         else:
-            retkey, retval = run(fdg_file)
+            retkey, retval = call_function(fdg_file, fdg_version=fdg_version)
             ret[retkey] = retval
     return ret
 
 
-def _get_top_data(topfile):
+def _get_top_data(topfile, fdg_version):
     """
     Helper method to retrieve and parse the FDG topfile
     """
     if not topfile.startswith('salt://'):
-        topfile = 'salt://' + BASE_DIR_FDG_PROFILES + os.sep + topfile
+        topfile = 'salt://' + BASE_DIR_FDG_PROFILES.get(fdg_version) + os.sep + topfile
```

Comment:
Continuing for input validation comment.
In that case, BASE_DIR_FDG_PROFILES.get(fdg_version) will be  None

Reply:
handled input validation